### PR TITLE
Add default level parameter in measure.find_contours

### LINKED ISSUE #4813: Make level parameter optional in measure.find_contours
## Description

I think that we could put a default value to level in find_contours. I would suggest (image.max() - image.min())/2. This would make this function able to be used with an image and I think this default value is meaningful.

I'm open to discussion.

### PR DESCRIPTION
**Pull request by rubywerman**
## Description
Fixes #4813
This PR adds the median as the default level parameter in `find_contours`. After reading the discussion in #4813 and looking at different image arrays [see [notebook](https://github.com/rubywerman/notebooks/blob/master/plot_contours.ipynb)], I thought the median could be a good default value. 

## Checklist

- [Docstrings for all functions](https://github.com/numpy/numpy/blob/master/doc/example.py)
- Gallery example in `./doc/examples` (new features only)
- Benchmark in `./benchmarks`, if your changes aren't covered by an
  existing benchmark
- Unit tests
- Clean style in [the spirit of PEP8](https://www.python.org/dev/peps/pep-0008/)

## For reviewers

- Check that the PR title is short, concise, and will make sense 1 year
  later.
- Check that new functions are imported in corresponding `__init__.py`.
- Check that new features, API changes, and deprecations are mentioned in
      `doc/release/release_dev.rst`.

### CODE REVIEW

File: skimage/measure/_find_contours.py [hunk 1]
```diff
@@ -9,7 +9,7 @@
 
 
 @deprecate_kwarg({'array': 'image'}, removed_version="0.20")
-def find_contours(image, level,
+def find_contours(image, level='median',
```
Comment (rfezzani) [inline skimage/measure/_find_contours.py:12]: Setting the default to a string may suggest that other string values are accepted ('mean' for example)
```suggestion
def find_contours(image, level=None,
```

**Comment by rubywerman** [comment]
Thank you for your feedback @rfezzani! I updated the parameter as you suggested :)

**Comment by emmanuelle** [comment]
Thanks for the PR @rubywerman! The discussion about which default level to choose was not completely settled in #4813, there is also the possibility in taking (max - min)/2. The latter choice is mostly useful for binary arrays while the median makes more sense for float arrays. I'd argue that binary arrays are the typical case where one does not want to specify the value of the level, because there is really only one value which makes sense. So I have a slight preference for (max - min)/2 but the median also makes sense.

Please also add a test which executes `find_contours` without passing the level value, so that the new API is tested.

**Comment by rubywerman** [comment]
Thanks for the feedback @emmanuelle! I will add the test. Also, could it be beneficial to have two default values, depending on whether it's a binary or float array?

**Comment by jni** [comment]
I agree with @sciunto that having different behaviour based on the input type isn't ideal. I have commented on the original issue, but my suggestion would be to have `np.mean(np.quantile(image, (0.005, 0.995)))` as the default. Thoughts?

**Comment by emmanuelle** [comment]
I have reconsidered my initial doubts. Given the fact that there is the possibility to pass a value, and that for a binary image other contours than 0.5 (in fact anything strictly between 0 and 1) will also good give results, we have to accept that there are several possible good solutions and just choose one of them. The one you suggest @jni is a good one, in fact the original median value proposed by @rubywerman is also good (and simple to explain).

File: skimage/measure/tests/test_find_contours.py [hunk 2]
```diff
@@ -130,3 +130,40 @@ def test_invalid_input():
         find_contours(r, 0.5, 'foo', 'bar')
     with testing.raises(ValueError):
         find_contours(r[..., None], 0.5)
+
```
Comment (emmanuelle) [inline skimage/measure/tests/test_find_contours.py:133]: thanks for adding all these tests :-)

File: skimage/measure/_find_contours.py [hunk 3]
```diff
@@ -24,7 +24,8 @@ def find_contours(image, level,
     image : 2D ndarray of double
         Input image in which to find contours.
     level : float
```
Comment (sciunto) [inline skimage/measure/_find_contours.py:26]:
```suggestion
    level : float, optional
```

File: skimage/morphology/convex_hull.py [hunk 4]
```diff
@@ -128,8 +128,8 @@ def convex_hull_object(image, *, connectivity=2):
     objects may overlap in the result. If this is suspected, consider using
     convex_hull_image separately on each object or adjust ``connectivity``.
     """
-    if image.ndim > 2:
-        raise ValueError("Input must be a 2D image")
+    if image.ndim > 3:
+        raise ValueError("Input must be a 2D or 3D image")
```
Comment (sciunto) [inline skimage/morphology/convex_hull.py:132]: Why this change in this PR?

Comment (rubywerman) [reply skimage/morphology/convex_hull.py:132]: my mistake– it shouldn't. fixing now

Comment (sciunto) [reply skimage/morphology/convex_hull.py]: Thank you!

**Comment by sciunto** [comment]
I approve it as well, I'm just waiting for the CI. Thank you for your contribution.

File: skimage/morphology/convex_hull.py (hunk 4 above)
Comment (emmanuelle) [reply skimage/morphology/convex_hull.py]: I thought @rubywerman was fixing #4865 at the same time :-) (but it can be in another PR, with a test ;-))

**Comment by emmanuelle** [comment]
Merging, thank you @rubywerman !
